# Lab 3 - Introduction to `np.linalg` Sub-Library

### Objective
Explore the basic functionalities of the `np.linalg` sub-library, including solving systems of equations, finding determinants, computing eigenvalues and eigenvectors, and performing matrix decompositions.

In [2]:
import numpy as np

### Task 1 - Solving Systems of Linear Equations
1. Consider the system of equations

    `3x + 4y = 7`<br />
    `2x - y = 1`
    1. Let the system be represented as a matrix equation in the form of $AX = B$. Solve for the vector `X` ($x$ and $y$).

In [3]:
# 1.
A = np.array([[3, 4], [2, -1]])
B = np.array([[7], [1]])

solution = np.linalg.solve(A, B)
print(solution, end="\n\n")

[[1.]
 [1.]]



### Questions
1. What are the values of $x$ and $y$?

In [4]:
# 1.
print(f'x = {solution[0]}')
print(f'y = {solution[1]}')

x = [1.]
y = [1.]


### Task 2 - Determinants

1. Calculate the determinant of the matrix `A`
$$A = \begin{bmatrix} 2 & 1 \\ 4 & -3 \end{bmatrix}$$

In [5]:
det = np.linalg.det(A)
print(f'Determinant = {det}')

Determinant = -11.000000000000002


### Questions

1. What is the value of the determinant?
2. What does the determinant tell you about the system of equations?

1. The value of the determinant is -11.000000000000002
2. The determinant tells you whether a system has a unique solution or not.
    1. If it is 0, there is no unique solution, otherwise there is